interfaces/builtin/usb_raw.go

```diff
@@ -0,0 +1,45 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const usbrawConnectedPlugAppArmor = `
+# Description: Allow raw access to all connected USB devices.
+# Reserved because this gives privileged access to the system.
+# Usage: reserved
+/dev/bus/usb/[0-9][0-9][0-9]/[0-9][0-9][0-9] rw,
+
+# Allow detection of usb devices. Leaks plugged in USB device info
+/sys/bus/usb/devices/ r,
+/sys/devices/pci**/usb[0-9]**/idVendor r,
+￼/sys/devices/pci**/usb[0-9]**/idProduct r,
+`
```

Comment:
@morphis, this is a powerful interface but the requested additional access are all related to usb devices and read-only, so yes, these are ok. That said, seeing the udev rule suggests to me that we should change `/run/udev/data/c189:[0-9] r, # USB serial converters` to:
```
/run/udev/data/c16[67]:[0-9] r, # ACM USB modems
/run/udev/data/b180:[0-9]    r, # various USB block devices
/run/udev/data/c180:[0-9]    r, # various USB character devices
/run/udev/data/c18[89]:[0-9] r, # USB serial converters
```